### DISCUSSION TITLE
Choosing a radio for GFSK on 435MHz

### DISCUSSION THREAD

**Discussion by manforowicz** [jgromes/RadioLib#901] [Q&A] [answered]
Hello! I'm on a project that's building a low-cost radio for a CubeSat. We'll put a transceiver IC that interfaces with RadioLib, onto a custom PCB that amplifies its output to at least 5 watts.

The amount of suitable transceiver ICs is overwhelming. We're looking for any advice about selecting one of them. Criteria:

- Can transmit and receive GFSK on 435-438 MHz.
- Has a minimal number of required external PCB components.
- Is simple to program and configure with RadioLib.

Currently we're leaning towards the SX1276, but are unsure if there are better options.

**Comment by jgromes** [comment, marked as answer]
All of the sub-GHz transceivers supported by RadioLib will fit the criteria, namely:

* CC1101
* SX123x/RF69
* Si443x/RFM2x
* SX127x
* SX126x

Here's a couple more things I would use when deciding:
1. The power consumption. Especially in Rx mode, since your radio will most likely always be receiving. Out of experience, power budget of nano- and pico-satellites is tight.
2. Flight heritage. Since you most likely don't plan on doing any radiation testing, you should find out which of these transceivers were flown already. I think that CC1101 and SX127x has been flown on a couple missions, and I know for a fact that SX126x was also flown.
3. Receiver sensitivity. Better sensitivity means at the very least more margin in the link budget, which can be exploited for higher data rate.
4. Data rate. Some of the transceivers support different data rates than others. I don't know what rate are you aiming for, though the common ones (4k8, 9k6 etc.) are supported by all of them.
5. TCXO support. SX126x has "native" TCXO support (as in, the IC itself can generate voltage reference for the TCXO). Otherwise you will need additional parts to support the TCXO. Or, if you have a detailed thermal analysis that shows the radio has a stable temperature, you can use a plain XTAL, calculate the frequency error due to temperature changes and see if that stil works. I would advise to use a TCXO though.

**Reply by manforowicz** [reply]
Thanks for the helpful information! We'll be sure to take these into account, especially the native TCXO.

Thank you also for maintaining this great library. It'll help make our code less dependent on the specific IC we choose.
